### PR TITLE
CB-12018 : updated tests to function with jasmine instead of jasmine-node

### PR DESCRIPTION
**Pull request by audreyso**
### What does this PR do?
Updated tests to function with jasmine instead of jasmine-node

### Checklist
- [X ] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [X ] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [X ] Added automated test coverage as appropriate for this change.

…node

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "cordova": "./bin/cordova"
   },
   "scripts": {
-    "test": "node node_modules/jasmine-node/bin/jasmine-node --captureExceptions --color spec",
+    "test": "node node_modules/jasmine/bin/jasmine --captureExceptions --color spec",
     "cover": "node node_modules/istanbul/lib/cli.js cover --root src --print detail node_modules/jasmine-node/bin/jasmine-node -- spec"
```
Comment (vladimir-kotikov) [inline package.json]: @audreyso, looks like you'll also need to update `cover` command to use jasmine inatead of jasmine-node

File: package.json [hunk 2]
```diff
@@ -31,6 +31,7 @@
     "cordova-common": "1.5.x",
     "cordova-lib": "6.4.0",
     "insight": "~0.8.2",
+    "jasmine": "^2.5.2",
```
Comment (vladimir-kotikov) [inline package.json]: Shouldn't this be a dev dependency?

File: spec/cli.spec.js [hunk 3]
```diff
@@ -55,76 +55,76 @@ describe("cordova cli", function () {
...
-                });
-            });
-
-            it("will spit out the version with --version", function (done) {
-                cli(["node", "cordova", "--version"], function () {
-                    expect(logger.results.mostRecentCall.args[0]).toMatch(version);
-                    done()
-                });
-            });
+        beforeEach(function () {
```
Comment (stevengill) [inline spec/cli.spec.js]: can probably delete this beforeEach

File: spec/cli.spec.js [hunk 4]
```diff
@@ -134,11 +134,11 @@ describe("cordova cli", function () {
 
     describe("create", function () {
         beforeEach(function () {
-            spyOn(cordova.raw, "create").andReturn(Q());
-            spyOn(cordova_lib, "CordovaError");
+            spyOn(cordova.raw, "create").and.returnValue(Q());
+            // spyOn(cordova_lib, "CordovaError");
```
Comment (stevengill) [inline spec/cli.spec.js]: do you remember why you commented this out?

File: package.json [hunk 5]
```diff
@@ -11,8 +11,8 @@
     "cordova": "./bin/cordova"
   },
   "scripts": {
-    "test": "node node_modules/jasmine-node/bin/jasmine-node --captureExceptions --color spec",
-    "cover": "node node_modules/istanbul/lib/cli.js cover --root src --print detail node_modules/jasmine-node/bin/jasmine-node -- spec"
+    "test": "jasmine --captureExceptions --color",
+    "cover": "jasmine"
```
Comment (vladimir-kotikov) [inline package.json]: Does Jasmine generate coverage reports? Shouldn't this line be the same as [in apache/cordova-lib#510](https://github.com/apache/cordova-lib/pull/510/files#diff-cc6b3b84bb015909917e7c201a31f65aR23)

File: spec/cli.spec.js [hunk 6]
```diff
@@ -55,76 +55,76 @@ describe("cordova cli", function () {
     });
 
     describe("options", function () {
-        describe("version", function () {
-            var version = require("../package").version;
+      describe("version", function () {
```
Comment (vladimir-kotikov) [inline spec/cli.spec.js]: nit: could you please use 4 spaces for indentation to be consistent with the rest of the tests/code?

File: package.json (hunk 5 above)
Comment (audreyso) [inline package.json]: Thank you! Updated to ...  ``` "cover": "istanbul cover --root src --print detail jasmine" ```

File: spec/cli.spec.js (hunk 4 above)
Comment (audreyso) [inline spec/cli.spec.js]: yes @stevengill ... it gives this error
```     Error: <spyOn> : CordovaError is not declared writable or has no setter
    Usage: spyOn(<object>, <methodName>) ```
Is this line still needed? I tried using this to fix it, but that did not work.
``` Object.defineProperty(cordova_lib, "CordovaError", { writable: true }); ```

I've seen some people fix it like this too, but not sure if this is correct.

```   var descriptor = Object.getOwnPropertyDescriptor(obj, methodName);
 if (!(descriptor.writable || descriptor.set)) {
throw new Error(methodName + ' is not declared writable or has no setter');
} ```

**Comment by stevengill** [comment]
@vladimir-kotikov hey! You make a good point about it being annoying to add tests later and have to update all the numbers. And I agree the stack trace is more useful. We ran into issues where the stack trace wasn't useful (especially with Q errors). That is why Audrey started labeling tests. 

I'm not too concerned with the numbers staying in order. If it does become annoying, we will go remove it. 

Appreciate your feedback :)

**Comment by vladimir-kotikov** [comment]
Yeah, not a big deal and probably not worth renaming all test cases back now, It just reminded me file transfer tests where we have 'spec.11' then `spec.9` then `spec.10`, etc. :)

**Pull request closed without merging**